test/unit/math/prim/prob/categorical_test.cpp

```diff
@@ -138,3 +149,40 @@ TEST(ProbDistributionsCategorical, chiSquareGoodnessFitTest) {
 
   EXPECT_TRUE(chi < quantile(complement(mydist, 1e-6)));
 }
+
+TEST(ProbDistributionsCategorical, categorical_vecRNG) {
+  using stan::math::categorical_rng;
+  using stan::math::softmax;
+  boost::random::mt19937 rng;
+
+  Eigen::VectorXd vec1(4);
+  vec1 << softmax(Eigen::VectorXd::Random(4));
+  Eigen::VectorXd vec2(4);
+  vec2 << softmax(Eigen::VectorXd::Random(4));
+  Eigen::VectorXd vec3(4);
+  vec3 << softmax(Eigen::VectorXd::Random(4));
+
+  std::vector<Eigen::VectorXd> vecs{vec1, vec2, vec3};
+
+  std::vector<int> rng_stvec = categorical_rng(vecs, rng);
+  for (int i = 0; i < 3; ++i) {
+    EXPECT_EQ(rng_stvec[i], categorical_rng(vecs[i], rng));
```

Comment:
How does this test work? It's generating the numbers once and then generating them again and making sure they're equal?

Reply:
Yep, making sure that the vectorised rng returns the same result as looping over the inputs

Reply:
I guess I'm not sure how it does work though -- like I don't see the rng seed getting reset and those vectors are random going in.